File: ruby/bob/bob.rb
require 'pp'

class Bob
  def hey (text)
    return 'Fine. Be that way!' if (text =~ /\A\s*\z/)
    return 'Whoa, chill out!' if is_forcefully(text) 
    return 'Sure.' if is_question(text)
    return 'Whatever.'
  end

  private

  def is_question (text)
    return true if (text =~ /\?\z/)
    false
  end

  def is_forcefully (text)
    return true if (text =~ /\!/ && text == text.upcase)
    return true if (text == text.upcase && not(text =~ /\d/))
    false
  end

end

File: lua/bob/bob.lua
bob = {}

local function is_silent(text) 
    return text == nil or string.len(text) == 0
end

local function is_shouting(text) 
    return string.upper(text) == text
end

local function is_question(text)
    return string.sub(text, -1) == '?'
end

function bob.hey(text)
  if is_silent(text)   then return 'Fine, be that way.' end
  if is_shouting(text) then return 'Whoa, chill out!'   end
  if is_question(text) then return 'Sure'               end
  return 'Whatever'
end  

return bob
